### PR TITLE
Create key pair if it doesn't exist

### LINKED ISSUE #43: Update EC2 plugin to provision EC2 key pair
## Overview

To fully deploy the EC2 plugin this will not require infra to be provisioned as EC2 plugin is responsible for creating it's required resources, however it will need to also provision an EC2 key pair that matches the supplied SSH key.

## Acceptance Criteria

- [ ] Upsert the SSH key provided as a EC2 key pair as part of the executor
- [ ] Update functional test pipeline to support changes

### PR DESCRIPTION
**Pull request by kessler-frost**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

Resolves #43

### CODE REVIEW

File: covalent_ec2_plugin/ec2.py [hunk 1]
```diff
@@ -54,6 +54,12 @@
     }
 )
 
+EC2_KEYPAIR_NAME = "covalent-ec2-keypair"
```
Comment (AlejandroEsquivel) [inline covalent_ec2_plugin/ec2.py:57]: Can we maybe add the term `executor` here for verboseness @kessler-frost 🙏🏼  
```suggestion
EC2_KEYPAIR_NAME = "covalent-ec2-executor-keypair"
```

File: covalent_ec2_plugin/infra/variables.tf [hunk 2]
```diff
@@ -19,7 +19,7 @@
 # Relief from the License may be granted by purchasing a commercial license.
 
 variable "name" {
-  default     = "covalent-svc"
+  default     = "covalent-ec2"
```
Comment (AlejandroEsquivel) [inline covalent_ec2_plugin/infra/variables.tf:22]: Same here
```suggestion
  default     = "covalent-executor-ec2"
```

File: covalent_ec2_plugin/ec2.py [hunk 3]
```diff
@@ -168,21 +176,45 @@ async def setup(self, task_metadata: Dict) -> None:
...
+        self.ssh_key_file = str(ec2_ssh_dir / f"{self.key_name}.pem")
+
+        # Try to import the key pair/ssh key file that might've been created earlier
+        # If those don't exist, create the key pair and save the key material to the ssh_key_file
+        if not Path(self.ssh_key_file).exists():
+            try:
+                key_pair = ec2.create_key_pair(KeyName=self.key_name)
+            except Exception as e:
+                if e.response["Error"]["Code"] != "InvalidKeyPair.Duplicate":
+                    raise e
```
Comment (AlejandroEsquivel) [inline covalent_ec2_plugin/ec2.py:205]:
```suggestion
                    raise
```

**Review by AlejandroEsquivel — CHANGES_REQUESTED** [review]
Overall looks great @kessler-frost  🙏🏼  some minor comments

**Comment by kessler-frost** [comment]
@AlejandroEsquivel made the changes ✅ !

**Review by AlejandroEsquivel — APPROVED** [review]
Awsome! thanks @kessler-frost

**Pull request merged**